### PR TITLE
Build miner inside Docker container for aarch64

### PR DESCRIPTION
**Pull request by fvasquez**
The `Dockerfile` builds and runs miner inside a container using QEMU.

### CODE REVIEW

**Comment by mrallen1** [comment]
Looks good. Tomorrow let's work on getting this running in buildkite

**Comment by mrallen1** [comment]
Despite the tests failure, I believe this PR is ready for review and merging. @Vagabond @evanmcc

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -5,7 +5,36 @@ steps:
...
+#
+#  - if: build.tag != null
+#    name: ":whale: ARM32 docker"
+#   agents:
+#     queue: "erlang"
+#   commands:
+#     - "git fetch -t"
+#     - ".buildkite/scripts/make_arm32_docker.sh"
+
+  - if: build.tag != null
```
Comment (madninja) [inline .buildkite/pipeline.yml:34]: is this section duplicated?

File: .buildkite/scripts/Dockerfile-amd64 [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+WORKDIR /opt/miner
+
+ENV COOKIE=miner \
+    # Write files generated during startup to /tmp
+    RELX_OUT_FILE_PATH=/tmp
+
+COPY --from=builder /opt/rel /opt/miner
+
+#CMD tail -f /dev/null
```
Comment (madninja) [inline .buildkite/scripts/Dockerfile-amd64:40]: Should the docker file strip out the source or other build shrapnel after building a release?

File: .buildkite/scripts/Dockerfile-arm32 [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+RUN ulimit -n 64000
+
+WORKDIR /opt/miner
+
+ENV COOKIE=miner \
+    # Write files generated during startup to /tmp
+    RELX_OUT_FILE_PATH=/tmp
+
+COPY --from=builder /opt/rel /opt/miner
+
```
Comment (madninja) [inline .buildkite/scripts/Dockerfile-arm32:41]: Same question? Does cleaning up a lot of the source and build output help make the image slimmer?

File: .buildkite/scripts/Dockerfile-arm64 [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+WORKDIR /opt/miner
+
+ENV COOKIE=miner \
+    # Write files generated during startup to /tmp
+    RELX_OUT_FILE_PATH=/tmp
+
+COPY --from=builder /opt/rel /opt/miner
+
+#CMD tail -f /dev/null
```
Comment (madninja) [inline .buildkite/scripts/Dockerfile-arm64:40]: Same question on retaining build/ source shrapnel after the docker build

File: .buildkite/pipeline.yml [hunk 5]
```diff
@@ -5,7 +5,36 @@ steps:
       - "make compile"
       - "make ci"
     name: ":hammer: build"
+    key: "tests"
```
Comment (madninja) [inline .buildkite/pipeline.yml:8]: What is this key for?

Comment (mrallen1) [reply .buildkite/pipeline.yml:8]: Well, eventually I'd like to make building the image on having a clean test run, so this just sets the stage for that. Down the road we can add a

```yaml
  depends_on: tests
```

line in there.

File: .buildkite/pipeline.yml (hunk 1 above)
Comment (mrallen1) [reply .buildkite/pipeline.yml:34]: Yes, so the builds will run in parallel with each other and only be trigged by a tag

File: .buildkite/scripts/Dockerfile-amd64 (hunk 2 above)
Comment (mrallen1) [reply .buildkite/scripts/Dockerfile-amd64:40]: This `COPY` command basically leaves all the build foo in a different container but moves our release into a fresh container.

File: .buildkite/scripts/Dockerfile-arm32 (hunk 3 above)
Comment (mrallen1) [reply .buildkite/scripts/Dockerfile-arm32:41]: Yes it does

File: .buildkite/pipeline.yml (hunk 1 above)
Comment (madninja) [reply .buildkite/pipeline.yml:34]: Right but it's building the same make_amd64_docker.sh twice.. in parallel

File: .buildkite/scripts/Dockerfile-arm64 (hunk 4 above)
Comment (mrallen1) [reply .buildkite/scripts/Dockerfile-arm64:40]: Yeah, tbh we should probably purge the build images from bk - that's a good clean up step.

Comment (mrallen1) [reply .buildkite/scripts/Dockerfile-arm64:40]: Honestly, we should probably just purge images in general after... 10 days maybe?

File: .buildkite/pipeline.yml (hunk 1 above)
Comment (madninja) [reply .buildkite/pipeline.yml:34]: I can't read worth crap today.

**Pull request merged**